recipes/dhcpcd/files/init

```diff
@@ -16,7 +16,12 @@ do_start() {
 }
 
 do_stop() {
-    start-stop-daemon -K --pidfile $PIDFILE --exec $DAEMON
+    # only try to stop the daemon if there is actually something
+    # running. Otherwise a restart before a start will fail
+    start-stop-daemon -K -o --pidfile $PIDFILE --exec $DAEMON
+    if [ "$?" != "0" ] ; then
+	start-stop-daemon -K --pidfile $PIDFILE --exec $DAEMON
+    fi
 }
```

Comment:
it does work, since the -o just "checking if work is to be done" according to the docs, which means we only really stop it if there is something to stop. It is surely not the most obvious way to solve this, but dropping set -e means we need to change it a lot more to handle errors